gensim/models/keyedvectors.py

```diff
@@ -311,11 +321,18 @@ def doesnt_match(self, words):
         """
         self.init_sims()
 
-        words = [word for word in words if word in self.vocab]  # filter out OOV words
-        logger.debug("using words %s" % words)
         if not words:
             raise ValueError("cannot select a word from an empty list")
-        vectors = vstack(self.syn0norm[self.vocab[word].index] for word in words).astype(REAL)
+        logger.debug("using words %s" % words)
+        vectors = []
+        for word in words:
+            try:
+                vectors.append(self.word_vec(word))
+            except KeyError:
+                logger.debug("vector for word %s not present, ignoring the word", word)
+        if not vectors:
+            raise ValueError("vector for all given words absent")
```

Comment:
Confusing message, please rephrase. How about `"cannot compute similarity with no input"` like above, for consistency?

Also wondering whether we should throw an exception in case of missing words, rather than silently ignoring them (DEBUG level message is almost ignore).